### PR TITLE
feat(ssh_tunnel): SSH Tunnel Switch extension

### PR DESCRIPTION
**Pull request by Antonio-RiveroMartnez**
### SUMMARY

This PR doesn't change the behavior of our UI. Originally, we had the switch and the rest of the form fields as part of the SSHTunnelForm, however, that stops us from extending the behavior of the switch itself if needed. With these changes we introduce a new registry extension and make use of it if provided. Also we add a new test to cover such scenario.

### TESTING INSTRUCTIONS
1. Implement a new extension and place it in the new registry
2. Provide such extension to the frontend
3. Your extension of the switch should be displayed and not the default one

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [X] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by eschutho — APPROVED** [review]
LGTM

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 1]
```diff
@@ -1346,14 +1357,33 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
...
       }
-      removeSSHTunnelConfig={() =>
-        setDB({
-          type: ActionType.removeSSHTunnelConfig,
-        })
-      }
     />
   );
 
+  const renderDefaultSSHTunnelSwitch = () => (
```
Comment (eric-briscoe) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:1363]: @Antonio-RiveroMartnez one improvement we could consider is to converting renderDefaultSSHTunnelSwitch function into an actual component in separate file.  Then import it at top of file.  Also add a prop which is `isSSHTunneling` where the switch component just returns undefined so the show / hide logic is internal to the switch.  Then we can collapse the conditional rendering blocks into a simple normal component block.

add the isSSHTunneling to interface:
````
export interface SwitchProps {
  isEditMode: boolean;
  isSSHTunneling: boolean;
  dbFetched: any;
  disableSSHTunnelingForEngine?: boolean;
  useSSHTunneling: boolean;
  setUseSSHTunneling: React.Dispatch<React.SetStateAction<boolean>>;
  setDB: React.Dispatch<any>;
}
```

Then where we set the SSHTunnelSwitchExtension we could change to something like:

```
import SSHTunnelSwitch from './SSHTTunnelSwitch';
const SSHTunnelSwitchComponent= extensionsRegistry.get(
    'ssh_tunnel.form.switch',
  ) ?? SSHTunnelSwitch;
```

this way lower in the code on lines 1591-1601 and 1877-1897
can look like:
```
<SSHTunnelSwitchComponent
  isEditMode={isEditMode}
  isSSHTunneling={isSSHTunneling}
  dbFetched={dbFetched}
  disableSSHTunnelingForEngine={disableSSHTunnelingForEngine}
  useSSHTunneling={useSSHTunneling}
  setUseSSHTunneling={setUseSSHTunneling}
  setDB={setDB}
/>

```
and not need to do all the condition checks if using extension vs base component, etc.  what do you think?

**Review by eric-briscoe — COMMENTED** [review]
@Antonio-RiveroMartnez Overall looks great!  I left one comment for potential simplification in the rendering logic for the switch.  code will work fine as is, but I think this would make it a lot more clear and simple.  Please let me know what you think.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx (hunk 1 above)
Comment (Antonio-RiveroMartnez) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:1363]: Oh! Yeah, that makes sense. Let me work on that and update the PR. Will come back to you when ready 👍 Thanks!

**Review by eric-briscoe — APPROVED** [review]
Looks great, thanks for the changes!

**Pull request merged**